### PR TITLE
Max collector should ignore only no result errors

### PR DESCRIPTION
**Pull request by arjunrn**
PR #47 has faulty logic. All errors including ones where the _metrics-adpater_ cannot connect to _prometheus_ are ignored. Only invalid results should be so if there is a connection error then there is no result computed.

### CODE REVIEW

**Comment by mikkeloscar** [comment]
:+1:

File: pkg/collector/prometheus_collector.go [hunk 1]
```diff
@@ -16,6 +16,14 @@ import (
 	"k8s.io/metrics/pkg/apis/custom_metrics"
 )
 
+type noResultError struct {
```
Comment (mikkeloscar) [inline pkg/collector/prometheus_collector.go]: nit: Probably makes sense to expose the type as it's returned by the public `GetMetrics` method.

**Comment by mikkeloscar** [comment]
:+1:

**Comment by arjunrn** [comment]
👍

**Pull request merged**